### PR TITLE
326942: Updated environments permissions

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items:*

[AB#326942](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/326942)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
image

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:

### CODE REVIEW

File: src/ADP.Portal.Api/Models/Ado/AdoGroup.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using Newtonsoft.Json;
```
Comment (kbabigumira) [inline src/ADP.Portal.Api/Models/Ado/AdoGroup.cs:1]: Why are we still using newtonsoft instead of System.Text.Json?

Comment (rajeshkaremane) [reply src/ADP.Portal.Api/Models/Ado/AdoGroup.cs]: Removed the unnecessary code

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            this.logger = logger;
+            adoOrgUrl = configuration.Value.OrganizationUrl;
+            client = new HttpClient();
+            client.DefaultRequestHeaders.Accept.Add(
+                    new MediaTypeWithQualityHeaderValue("application/json"));
+
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                Convert.ToBase64String(
+                    System.Text.Encoding.ASCII.GetBytes(
+                        string.Format("{0}:{1}", "", configuration.Value.PatToken))));
```
Comment (danny-may) [inline src/ADP.Portal.Api/Services/AdoRestAPIService.cs:27]: {0} is always an empty string here, is that intentional?
Maybe using an interpolated string would be better?
`$":{configuration.Value.PatToken}"`

**Review by danny-may — CHANGES_REQUESTED** [review]

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs (hunk 2 above)
Comment (rajeshkaremane) [reply src/ADP.Portal.Api/Services/AdoRestAPIService.cs:27]: Yes it is intentional. Thats how the auth is constructed for ADO

**Review by danny-may — APPROVED** [review]

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
+﻿using ADP.Portal.Api.Config;
+using ADP.Portal.Api.Models.Ado;
+using ADP.Portal.Core.Ado.Infrastructure;
+using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+
+namespace ADP.Portal.Api.Services
```
Comment (ppotturi) [inline src/ADP.Portal.Api/Services/AdoRestAPIService.cs:7]: I am not sure why the services are created in the API project ? if it is a business/repostiry layer should move to the core project.  Can we discuss on this?

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs (hunk 2 above)
Comment (ppotturi) [reply src/ADP.Portal.Api/Services/AdoRestAPIService.cs:26]: to simply, we can change it to interpolated string  $":{configuration.Value.PatToken}" will produce the same output as the string format.

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                Convert.ToBase64String(
+                    System.Text.Encoding.ASCII.GetBytes(
+                        string.Format("{0}:{1}", "", configuration.Value.PatToken))));
+        }
+
+
+
+        public async Task<string> GetUserIdAsync(string projectName, string userName)
+        {
+            var uri = adoOrgUrl.Replace("dev.azure.com", "vssps.dev.azure.com") + "/_apis/identities?searchFilter=General&filterValue=[" + projectName + "]\\" + userName + "&queryMembership=None&api-version=7.1-preview.1";
```
Comment (ppotturi) [inline src/ADP.Portal.Api/Services/AdoRestAPIService.cs:33]: there no libraries for these calls?

Comment (rajeshkaremane) [reply src/ADP.Portal.Api/Services/AdoRestAPIService.cs]: As discussed there is no sdk available for securityrole.

File: src/ADP.Portal.Api/Services/AdoRestAPIService.cs (hunk 3 above)
Comment (rajeshkaremane) [reply src/ADP.Portal.Api/Services/AdoRestAPIService.cs]: Updated code.

File: src/ADP.Portal.Core/Ado/Client/ADORestHttpClient.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            this.baseUrl = baseUrl;
+        }
+
+        public AdoRestHttpClient(Uri baseUrl, VssCredentials credentials, VssHttpRequestSettings settings, params DelegatingHandler[] handlers) : base(baseUrl, credentials, settings, handlers)
+
+        {
+            this.baseUrl = baseUrl;
+        }
+
+        public HttpClient getHttpClient()
```
Comment (danny-may) [inline src/ADP.Portal.Core/Ado/Client/ADORestHttpClient.cs:38]: Pascal case for method names - `GetHttpClient` & `GetOrganizationUrl`

File: src/ADP.Portal.Core/Ado/Infrastructure/AdoRestAPIService.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                        default:
+                            break;
+                    }
+                }
+            }
+            logger.LogInformation("Security Role List: {SecurityRoleList} ", adoSecurityRoleList.ToString());
+            return adoSecurityRoleList;
+        }
+
+        public async Task<bool> updateRoleAssignmentAsync(string projectId, string envId, List<AdoSecurityRole> adoSecurityRoleList)
```
Comment (danny-may) [inline src/ADP.Portal.Core/Ado/Infrastructure/AdoRestAPIService.cs:68]: Pascal case method name

**Review by danny-may — CHANGES_REQUESTED** [review]

File: src/ADP.Portal.Core/Ado/Infrastructure/AdoRestAPIService.cs (hunk 6 above)
Comment (rajeshkaremane) [reply src/ADP.Portal.Core/Ado/Infrastructure/AdoRestAPIService.cs]: Fixed

File: src/ADP.Portal.Core/Ado/Client/ADORestHttpClient.cs (hunk 5 above)
Comment (rajeshkaremane) [reply src/ADP.Portal.Core/Ado/Client/ADORestHttpClient.cs]: Fixed

**Review by ppotturi — APPROVED** [review]

**Review by danny-may — APPROVED** [review]

**Pull request merged**